src/server/main_service.cc

```diff
@@ -1244,6 +1244,7 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   DCHECK_NE(0u, shard_set->size()) << "Init was not called";
 
   ServerState& etl = *ServerState::tlocal();
+  DCHECK_EQ(etl.thread_index(), cntx->thread_index());
```

Comment:
we crash here during EvalOffloading because we run the command in another thread but pass ConnectionContext from the io thread.